test-framework/sudo-compliance-tests/src/sudo/sudoers/run_as.rs

```diff
@@ -451,3 +451,33 @@ fn supplemental_group_matching() -> Result<()> {
 
     Ok(())
 }
+
+/// This test tracks [CVE-2019-14287](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2019-14287)
+/// which is explained in more detail [here](https://www.sudo.ws/security/advisories/minus_1_uid/).
+#[test]
+fn minus_1_uid() -> Result<()> {
+    let host = "myhost";
+
+    let env = Env(format!(
+        "{USERNAME} {host} = (ALL, !root) NOPASSWD: /usr/bin/id"
+    ))
+    .user(User(USERNAME))
+    .group(GROUPNAME)
+    .hostname(host)
+    .build()?;
+
+    for uid in [(-1).to_string(), u32::MAX.to_string()] {
+        let output = Command::new("sudo")
+            .args(["-u", &format!("#{uid}"), "id", "-u"])
+            .output(&env)?;
+
+        assert!(!output.status().success());
+        if sudo_test::is_original_sudo() {
+            assert_contains!(output.stderr(), "unknown user");
```

Comment:
nit: you could refactor the `assert_contains` if you have the `if` expression return the "needle" string into a variable